package.json

```diff
@@ -30,7 +30,7 @@
     "docs:export": "next export -o docs/export",
     "prebuild": "rimraf build",
     "build:es2015": "cross-env NODE_ENV=production babel ./src --ignore *.spec.js --out-dir ./build",
-    "build:es2015modules": "cross-env NODE_ENV=production BABEL_ENV=modules babel ./src/index.js --out-file ./build/index.es.js",
+    "build:es2015modules": "cross-env NODE_ENV=production BABEL_ENV=modules babel ./src --out-dir ./build/es",
```

Comment:
We can't change this module, it's required to support old browser. Even react-redux is following our current pattern. The best we can do is ship another folder with a different babel-preset-env configuration.

Reply:
We could call this new script `build:esnext`

Reply:
This also means the `esnext:main` / `module` definitions in the `package.json` cannot be changed, right? Couldn't this fall under the breaking changes of v1?

Reply:
Looking at https://www.reddit.com/r/javascript/comments/5jwg9c/confused_about_fields_module_and_jsnextmain_in/

I think that we can remove the `esnext:main` key as it's non-standard.

> This also means the esnext:main / module definitions in the package.json cannot be changed, right? 

Ideally, we would have two key versions in the `package.json`.
One for people that don't transpile the node_modules. The `main` key sounds like the best option.
One for people that do transpile the node_modules. There is not option for this as far as I know in the modern building tools.

Also, given that transpiling the node_modules is an advanced mode of any building tool, I think simply having a folder for it is enough.

So now, comes the naming question of those folders. I think that we should keep the flat structure for our `es2015` version. What about using a direct naming convention. `es2015modules` for the `es2015modules` script and `es` for the new `es` script?

Reply:
Agh, I didn't see this comment here before doing the edits yesterday.

> Also, given that transpiling the node_modules is an advanced mode of any building tool, I think simply having a folder for it is enough.

Hmm, I assume that when somebody depends on the `module` field serving ES5 code with ES6 modules something is off. I would expect that the code thats listed under `module` to be _at least_ full ES6 code, if not ES6+.